sdk/src/wallet/wallet/operations/stronghold_backup/stronghold_snapshot.rs

```diff
@@ -77,23 +60,13 @@ pub(crate) async fn read_data_from_stronghold_snapshot(
     Option<SecretManagerDto>,
     Option<Vec<AccountDetails>>,
 )> {
-    migrate_backup(stronghold).await?;
+    migrate(stronghold).await?;
 
     // Get client_options
-    let client_options_bytes = stronghold.get(CLIENT_OPTIONS_KEY.as_bytes()).await?;
-    let client_options = if let Some(client_options_bytes) = client_options_bytes {
-        let client_options_string = String::from_utf8(client_options_bytes)
-            .map_err(|_| crate::wallet::Error::Backup("invalid client_options"))?;
-        let client_options: ClientOptions = serde_json::from_str(&client_options_string)?;
-
-        log::debug!("[restore_backup] restored client_options {client_options_string}");
-        Some(client_options)
-    } else {
-        None
-    };
+    let client_options = stronghold.get::<ClientOptions>(CLIENT_OPTIONS_KEY).await?;
```

Comment:
Don't really need the explicit type (to be consistent with your restored_secret_manager get)